sdk/nodejs/runtime/closure/createClosure.ts

```diff
@@ -1058,7 +1058,7 @@ function getOrCreateEntry(
     }
 
     function captureModule(moduleName: string) {
-        if (moduleName[0] === ".") {
+        if (moduleName.startsWith(".") && !moduleName.startsWith("./node_modules/")) {
```

Comment:
Can we simplify this to just checking for `moduleName.startsWith("./node_modules/")` and doing what is currently the `else` branch below, and leave the "else" branch to cover the "." case?

Also - the `else if (moduleName.startsWith("@pulumi")) {` is probably wrong too - right?  I assume this comes back from `findModuleName` as `./node_modules/@pulumi/pulumi`?

Reply:
right.  the @pulumi part is likely wrong.  i'll use this as the opportunity to add the "deploymentOnly" module bit.

Reply:
Done.